src/services/ContributorAuditService/providers/Bitbucket/BitbucketApiClient.ts

```diff
@@ -0,0 +1,248 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import { soosLogger } from "../../../../logging/SOOSLogger";
+import { DateUtilities, sleep } from "../../../../utilities";
+import { SOOS_BITBUCKET_CONTRIBUTOR_AUDIT_CONSTANTS } from "./constants";
+import {
+  IContributorAuditRepositories,
+  IContributorAuditRepository,
+} from "../../../../api/SOOSHooksApiClient";
+
+interface IBitbucketHttpRequestParameters {
+  baseUri: string;
+  username: string;
+  password: string;
+}
+
+interface IHttpClientParameters extends IBitbucketHttpRequestParameters {
+  apiClientName: string;
+  dateToFilter: string;
+}
+
+export interface BaseBitBucketApiResponse {
```

Comment:
need to clean all this type gymnastics (all of this is to try and implement a pagination filter that works for both)